tests/test_operation.py

```diff
@@ -1506,3 +1505,97 @@ def test_composed(self):
+        expected = np.array([[0, 0, 1, 0], [0, 0, 0, 1], [1, 0, 0, 0], [0, 1, 0, 0]])
+        assert np.allclose(expected, res)
+
+    def test_autograd(self, tol):
+        """Tests differentiation in autograd by checking how a specific element of the expanded matrix depends on the
+        canonical matrix."""
+
+        def func(mat):
+            res = qml.operation.expand_matrix(mat, wires=[2], wire_order=[0, 2])
+            return res[0, 1]
+
+        base_matrix = pnp.array([[0.0, 1.0], [1.0, 0.0]], requires_grad=True)
+        grad_fn = qml.grad(func)
+        gradient = grad_fn(base_matrix)
+
+        # the entry should propagate from position (0, 1) in the original tensor
+        expected = np.array([[0.0, 1.0], [0.0, 0.0]])
+        assert np.allclose(gradient, expected, atol=tol)
+
+    def test_torch(self, tol):
+        """Tests differentiation in torch by checking how a specific element of the expanded matrix depends on the
+        canonical matrix."""
+        torch = pytest.importorskip("torch")
+
+        base_matrix = torch.tensor([[0.0, 1.0], [1.0, 0.0]], requires_grad=True)
+        res = qml.operation.expand_matrix(base_matrix, wires=[2], wire_order=[0, 2])
+        element = res[0, 1]
+        element.backward()
+        gradient = base_matrix.grad
+
+        # the entry should propagate from position (0, 1) in the original tensor
+        expected = torch.tensor([[0.0, 1.0], [0.0, 0.0]])
+        assert np.allclose(gradient, expected, atol=tol)
+
+    def test_jax(self, tol):
+        """Tests differentiation in jax by checking how a specific element of the expanded matrix depends on the
+        canonical matrix."""
+        jax = pytest.importorskip("jax")
+        from jax import numpy as jnp
+
+        def func(mat):
+            res = qml.operation.expand_matrix(mat, wires=[2], wire_order=[0, 2])
+            return res[0, 1]
+
+        base_matrix = jnp.array([[0.0, 1.0], [1.0, 0.0]])
+        grad_fn = jax.grad(func)
+        gradient = grad_fn(base_matrix)
+
+        # the entry should propagate from position (0, 1) in the original tensor
+        expected = np.array([[0.0, 1.0], [0.0, 0.0]])
+        assert np.allclose(gradient, expected, atol=tol)
+
+    def test_tf(self, tol):
+        """Tests differentiation in TensorFlow by checking how a specific element of the expanded matrix depends on the
+        canonical matrix."""
+        tf = pytest.importorskip("tensorflow")
+
+        base_matrix = tf.Variable([[0.0, 1.0], [1.0, 0.0]])
+        with tf.GradientTape() as tape:
+            res = qml.operation.expand_matrix(base_matrix, wires=[2], wire_order=[0, 2])
+            element = res[0, 1]
+
+        gradient = tape.gradient(element, base_matrix)
+
+        # the entry should propagate from position (0, 1) in the original tensor
+        expected = tf.constant([[0.0, 1.0], [0.0, 0.0]])
+        assert np.allclose(gradient, expected, atol=tol)
```

Comment:
Tests look really thorough! A very minor suggestion, I wonder if you copy and paste the `utils.expand` tests here, if they will pass + maybe cover additional edge cases?